### ISSUE TITLE
You can start and finish do-afters while moving in space

### ISSUE THREAD

**Issue by Kayek14** [space-wizards/space-station-14#21109]
## Description
While moving in space with a jetpack or when near tiles, you can start and finish doafters (such as healing with bruise packs) even when changing moment with the directional keys. 

When using doafters while on a grid with gravity, any movement will cancel the effect.

**Reproduction**
1. Go into space
2. start healing yourself with a bruise pack
3. watch how you can move around and change direction without interupting the do-after

**Screenshots**

Using a bruise pack while moving near tiles:
![image](https://github.com/space-wizards/space-station-14/assets/106507028/8682b093-304d-48fd-a764-c56cf61ad106)

Using a bruise pack while moving with the jetpack:
![image](https://github.com/space-wizards/space-station-14/assets/106507028/9580ed83-9f29-4d48-bdb5-e9efc14ee314)

**Additional context**
Any movement should cancel a do-after. Not sure why it works while in space.

This seems to work with all do-afters that i tried

**Comment by deltanedas** [comment]
its intended so you can cuff and strip people while drifting in space as even the 0.01m/s movement would break doafters

might be worth disabling for anything where you are the doafter target if healing in space is that bad

**Comment by Kayek14** [comment]
Ah that makes sense. the being unable to cuff or do anything if grav was gone was annoying.

Healing in space is not really a big issue, but it makes it near impossible to kill someone if they have meds and a jetpack since they can run and heal at the same time

**Comment by deltanedas** [comment]
(((((((((((sosa))))))))))

**Comment by metalgearsloth** [comment]
Most do_afters should be distance based rather than movement based anyway, with the exception of like cuffing / force-feeding I think.

**Comment by Kayek14** [comment]
> (((((((((((sosa))))))))))

How did you know i was spectating them when i got the idea for this issue??

**Comment by deltanedas** [comment]
sosa 1v80 the station with bruise packs and jetpack for the 5th time in this month

**Comment by JoeHammad1844** [comment]
> Most do_afters should be distance based rather than movement based anyway, with the exception of like cuffing / force-feeding I think.

i disagree, i was playing goonstation where this is implemented and i got forced to become a fucking santa themed meat cube by the chaplain even though i moved during the (stupidly fast) do after

**Comment by deltanedas** [comment]
sacrificing(?) people would be lumped in with cuffing and force feeding

**Comment by JoeHammad1844** [comment]
he forced me to sign a paper

**Comment by MilenVolf** [comment]
Is this issue still valid?

**Comment by Dutch-VanDerLinde** [comment]
> Is this issue still valid?

yes

**Comment by FairlySadPanda** [comment]
I have triaged this bug. Impelling yourself in weightlessness scenarios should cancel do-afters, but simply moving at a high speed should not.
